src/expression.ts

```diff
@@ -247,7 +267,7 @@ export class PropAccessExpr extends BaseExpr<NodeKind.PropAccessExpr> {
      * Range of text in the source file where this Node resides.
      */
     span: Span,
-    readonly expr: Expr,
+    readonly expr: Expr | SuperKeyword,
```

Comment:
Hmm, thought I checked in to this but may have missed it. Super is not valid in all places that an Expression is but I agree it would clean things up to just treat it as an Expression. Will clean it up.

Reply:
Thanks. the change to make `CallExpr<E extends Expr | Super | This>` was one of the hotspots for typescript too. Minor though.

Reply:
Oh that's interesting. You'll have to walk me through how you analyzed the hot spots. That sounds very useful

Reply:
https://github.com/microsoft/TypeScript/wiki/Performance#performance-tracing

```console
$ tsc -p ./some/project/src/tsconfig.json --generateTrace tracing_output_folder
$ npx @typescript/analyze-trace tracing_output_folder
```

this will output a summary, looks like it was improved with 4.8 and gave some better updates.

Can also load the whole trace in `chrome://tracing`.

`node.ts` and `asl/synth.ts` are the worst right now.